Dockerfile

```
@@ -0,0 +1,30 @@
+FROM python:alpine as base
+
+EXPOSE 8000
+RUN apk add --no-cache \
+    curl \
+    npm \
+    nodejs
+RUN curl -sSL https://install.python-poetry.org | python3 -
+ENV PATH="/root/.local/bin:$PATH"
+
+WORKDIR /code
+COPY /package.json /code
+COPY /package-lock.json /code
+COPY /poetry.toml /code
+COPY /poetry.lock /code
+COPY /pyproject.toml /code
```

Although extra layers are a slight overhead, I'd definitely call this premature optimisation territory, but worth noting, since you may also find it easier to read, that we can combine these copys where the file name remains consistent inside the container: E.g.:
```suggestion
COPY /package.json /package-lock.json /code/
COPY /poetry.toml /poetry.lock /pyproject.toml /code/
```